### PR TITLE
block close channel when fee rate is too high

### PR DESCRIPTION
**Pull request by jjyr**

### CODE REVIEW

File: mutiny-core/src/nodemanager.rs [hunk 1]
```diff
@@ -1729,6 +1729,24 @@ impl<S: MutinyStorage> NodeManager<S> {
                         None
                     };
 
+                    // typical close tx size
+                    let close_tx_size = 345;
```
Comment (duanyytop) [inline mutiny-core/src/nodemanager.rs:1733]: It seems that many channels are closing trades with a size around 200

Comment (jjyr) [reply mutiny-core/src/nodemanager.rs:1733]: https://mempool.space/zh/testnet/tx/c68860ba8794a64907da8a175d71729e928b7b2430308cf44283548cc2f3497b

This is a close tx.

Comment (duanyytop) [reply mutiny-core/src/nodemanager.rs:1733]: I see.

**Review by duanyytop — APPROVED** [review]

**Review by EthanYuan — APPROVED** [review]

File: mutiny-core/src/nodemanager.rs (hunk 1 above)
Comment (Flouse) [reply mutiny-core/src/nodemanager.rs:1733]: > https://mempool.space/zh/testnet/tx/c68860ba8794a64907da8a175d71729e928b7b2430308cf44283548cc2f3497b

### In this tx
- Size: 345 bytes
- Virtual Size: 180 bytes
- Weight: 720 WU (weight units) = 4 * vByte

### Fee Rate
- 4,347 sat/vB
  fee = 4347 * 180 = 782,463 sats
- 4,347 * 250 = 1,086,750 sat/kw
  fee = 1,086,750 * 720 / 1000 = 782,460 sats

File: mutiny-core/src/nodemanager.rs [hunk 2]
```diff
@@ -1729,6 +1729,24 @@ impl<S: MutinyStorage> NodeManager<S> {
...
+                    let initiator_balance = if channel.is_outbound {
+                        channel.outbound_capacity_msat / 1000
+                    } else {
+                        channel.inbound_capacity_msat / 1000
+                    };
+                    if let Some(feerate_pkw) = target_feerate_sats_per_1000_weight {
+                        let estimate_fee: u64 = ((close_tx_size * feerate_pkw) / 1000).into();
```
Comment (ShookLyngs) [inline mutiny-core/src/nodemanager.rs:1740]: Can the byte length of the co-op close transaction (`close_tx_size`) be multiplied by the fee rate per kilo-weight (`feerate_per_kw`)? Perhaps it should use `tx_size_vbyte` and `feerate_per_vbyte` for the calculation instead:

```
close_tx_size_vbyte = 180
feerate_per_vbyte = feerate_pkw / 250
estimate_fee = close_tx_size_vbyte * feerate_per_vbyte
```

Another issue is the example you provided has two outputs are `P2WPKH` and `P2TR`. Here's another example where both outputs are `P2TR`, and its size in `vbyte` is larger (~192 vb) than your example (180 vb): [341e8bd017aa0ce8042d61784cb7bb81a3da1c0a945cf98ed24c28240390925f](https://mempool.space/testnet/tx/341e8bd017aa0ce8042d61784cb7bb81a3da1c0a945cf98ed24c28240390925f).

**Pull request merged**